web/src/lib/components/alert/alert.tsx

```diff
@@ -0,0 +1,49 @@
+import { h, Component } from 'preact';
+import Icon from '../icon';
+
+interface Props {
+  text: string;
+  active: boolean;
+}
+
+interface State {
+  active: boolean;
+}
+
+export default class Alert extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.onClick = this.onClick.bind(this);
+  }
+
+  state: State = {
+    active: false,
+  };
+
+  componentWillReceiveProps(nextProps: Props) {
```

Comment:
you can destructure `nextProps` like so:

```
componentWillReceiveProps({ active }: Props) {
  // you can use "active" here instead of "nextProps"
}
```

-> slightly less verbose :)

you can look at the changes in #578 for more examples of this style